cmd/namespace-lock_test.go

```diff
@@ -128,7 +140,51 @@ func TestNamespaceLockTest(t *testing.T) {
 	}
 }
 
-func TestLockStats(t *testing.T) {
+func TestNamespaceLockTimedOut(t *testing.T) {
+
+	// Get write lock
+	if !globalNSMutex.Lock("my-bucket", "my-object", "abc", 60*time.Second) {
+		t.Errorf("Failed to acquire lock")
+	}
+
+	// Second attempt for write lock on same resource should time out
+	locked := globalNSMutex.Lock("my-bucket", "my-object", "def", 1*time.Second)
+	if locked {
+		t.Errorf("Should not have acquired lock")
+	}
+
+	// Read lock on same resource should also time out
+	locked = globalNSMutex.RLock("my-bucket", "my-object", "def", 1*time.Second)
+	if locked {
+		t.Errorf("Should not have acquired read lock while write lock is active")
+	}
+
+	// Release write lock
+	globalNSMutex.Unlock("my-bucket", "my-object", "abc")
+
+	// Get read lock
+	if !globalNSMutex.RLock("my-bucket", "my-object", "ghi", 60*time.Second) {
+		t.Errorf("Failed to acquire read lock")
+	}
+
+	// Write lock on same resource should time out
+	locked = globalNSMutex.Lock("my-bucket", "my-object", "klm", 1*time.Second)
+	if locked {
+		t.Errorf("Should not have acquired lock")
+	}
+
+	// 2nd read lock should be just fine
+	if !globalNSMutex.RLock("my-bucket", "my-object", "nop", 60*time.Second) {
+		t.Errorf("Failed to acquire second read lock")
+	}
+
+	// Release both read locks
+	globalNSMutex.RUnlock("my-bucket", "my-object", "ghi")
+	globalNSMutex.RUnlock("my-bucket", "my-object", "nop")
+}
+
+// This test fails -- disabled for now
```

Comment:
This test case is flawed as it is making implicit assumptions about the order of events (plus explicit synchronisation) so it is actually a bit of surprise that it is working.
Removing this testcase (also in lieu of additional test cases in lsync).